scipio/src/executor.rs

```diff
@@ -1337,3 +1366,45 @@ fn test_no_spin() {
         "expected user time on LE is less than 1 millisecond"
     );
 }
+
+/// This test ensures following invariants:
+/// - executor does not spin when not asked
+/// - executor does spin when it is required only for requested time
+/// - executor run counts runtime duration for both queue and future
+#[test]
+fn test_spin() {
+    use crate::timer;
+
+    let dur = Duration::from_secs(1);
+    let ex0 = LocalExecutorBuilder::new().make().unwrap();
+    ex0.run(async { timer::sleep(dur).await });
+
+    let ex = LocalExecutorBuilder::new()
+        .spin_before_park(Duration::from_millis(10))
+        .make()
+        .unwrap();
+    let task = ex.spawn(async move { timer::sleep(dur).await });
+    ex.run(async { task.await });
+
+    let stats0 = &ex0.queues.borrow().stats;
```

Comment:
There should be a way to acquire the stats without looking into the queues.
The queues are an implementation detail, and from outside the executor should not be needed.

Reply:
Wiped